src/classes/CON_Email.cls

```diff
@@ -176,26 +178,29 @@ public class CON_Email {
 
             this.allFields = new List<CON_EmailField>();
             this.valuedFields = new List<CON_EmailField>();
-            Map<String, Schema.DescribeFieldResult> fieldMap = UTIL_Describe.getFieldsOfType('Contact', 'EMAIL');
-            this.emailLabel = fieldMap.get('Email').getLabel();
+
+            if (contactEmailFieldMap == null) {
+                contactEmailFieldMap = UTIL_Describe.getFieldsOfType('Contact', 'EMAIL');
+            }
+            this.emailLabel = contactEmailFieldMap.get('Email').getLabel();
 
             Map<String, String> hedaEmailField = new Map<String, String>();
             //build map for HEDA email field map (Label, API Name)
-            for(String fieldName : fieldMap.keySet() ) {
+            for(String fieldName : contactEmailFieldMap.keySet() ) {
                 if(String.isNotBlank(UTIL_Namespace.getNamespace()) && fieldName.startsWithIgnoreCase(UTIL_Namespace.getNamespace())) {
-                    hedaEmailField.put(fieldMap.get(fieldName).getLabel(), fieldName);
+                    hedaEmailField.put(contactEmailFieldMap.get(fieldName).getLabel(), fieldName);
                 }
             }
 
-            for(String fieldName : fieldMap.keySet() ) {
+            for(String fieldName : contactEmailFieldMap.keySet() ) {
                 //If the field is not a standard email field and it is not duplicate with HEDA email, then add it to the list
-                if ( fieldName != emailLabel && !(hedaEmailField.get(fieldMap.get(fieldName).getLabel()) != null &&
-                                fieldName != hedaEmailField.get(fieldMap.get(fieldName).getLabel()))) {
+                if ( fieldName != emailLabel && !(hedaEmailField.get(contactEmailFieldMap.get(fieldName).getLabel()) != null &&
```

Comment:
since these checks are longer, would you mind putting one condition per line:
`if(check 1 &&
     check2 && 
     check3) `
or something similar. Also there are a ton of parenthesis here maybe breaking out some of these checks into variables? I'm less concerned about this than I am the breaking up of the conditions though.

Reply:
I see my carriage returns didn't take above...those checks should have been like:

check1 &&
check2 &&
check3

Reply:
@lifewithryan Thanks - I've made the requested changes.